### PR TITLE
Add Protect API

### PR DESCRIPTION
**Pull request by andrewnicols**
Add an API call for the Protect API endpoint.
https://www.mediawiki.org/wiki/API:Protect

This takes all of the arguments available for the Protect API.

I have provided tests where possible, but unfortunately vows provides no way to mock the Client to do any more than the very basics. I've tested this locally with my own site, though I'm not sure of the format for the timestamp to be able to verify that (though it is just passing in the value).

I've opted to only support the alternative expiry format - that is that where multiple protections exist, each has its own expiry date.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -16,3 +16,4 @@ divdavem (https://github.com/divdavem)
 maxkueng (https://github.com/maxkueng)
 stdob (https://github.com/stdob)
 Technical-13 (https://github.com/Technical-13)
+Andrew Lyons (https://github.com/andrewnicols)
```
Comment (macbre) [inline AUTHORS:19]: :+1:

**Review by macbre — APPROVED** [review]

**Comment by macbre** [comment]
Thanks! I'm planning to migrate tests suite to jest. This will make mocking way easier.

**Comment by andrewnicols** [comment]
I shall look forward to it!

How often do you usually tag and release after new features land?

**Comment by macbre** [comment]
There hasn't been any active development in this repo recently. But I'll merge your changes soon and tag a release & publish it to npm.

Thanks again for your fixes :) Welcome on board.

**Pull request merged**